extensions/bitbucket/src/queries/index.ts

```diff
@@ -0,0 +1,71 @@
+import { Bitbucket } from 'bitbucket'
+import { preferences } from "../helpers/preferences";
+
+const clientOptions = {
+  baseUrl: 'https://api.bitbucket.org/2.0',
+  auth: {
+    username: preferences.accountName,
+    password: preferences.appPassword,
+  },
+  notice: false
+}
+
+const defaults = {
+  workspace: preferences.workspace,
+}
+
+const bitbucket = new Bitbucket(clientOptions)
+
+export async function getRepositories(): Promise<any> {
+  return await bitbucket.repositories.list({
+    ...defaults,
+    pagelen: 50,
```

Comment:
In my use case 50 is already to little. But I notice that the response time of the Bitbucket API also isn't to great. Getting pagelen 50 already takes >500ms and rises exponentially. Like you said in the readme.md we should probably fetch more repositories in the background and cache them for some amount of time as they don't change that often.

Reply:
I thought for a first release to set 50 as default and let the user change it as a preference text field, what do you think?

Reply:
I'm afraid that users will hurt themself by settings the `pagelen` to high. Also the API only accepts values until 100.
I was thinking, when the command is executed
- fetch repositories from storage
- every X amount of time update the storage cache by fetching all repositories.
- provide an action to refresh the cache (so fetch all repositories)

Maybe it's a good idea a do a pull request for this as well as I'm in the workspace with +2000 repos.

Reply:
Repositories are not changing a lot, so a cache makes sense.
Is it possible to run a process in the background with Raycast? Like a cronjob to fetch all repositories.
And to avoid spamming the bitbucket API we should run the fetch one by one, incrementing with the page number.

For a high number of repos, we can as well optimize the query to only return the fields we need.
Here I did it with the [PipelinesGetQuery() fields](https://github.com/Lp-Francois/extensions/blob/add-extension-bitbucket/extensions/bitbucket/src/queries/index.ts#L36) option.

[From bitbucket documentation: ](https://developer.atlassian.com/cloud/bitbucket/rest/intro/#partial-response)
> For better performance, you can ask the server to only return the fields you really need and to omit unwanted data. To request a partial response and to add or remove specific fields from a response, use the fields query parameter.